gammapy/background/models.py

```diff
@@ -71,3 +81,775 @@ def evaluate(self, x, y):
+        energy_bin, energy_bin_edges = self.find_energy_bin(energy)
+        ss_energy_bin_edges = "[{0}, {1}) {2}".format(energy_bin_edges[0].value,
+                                                      energy_bin_edges[1].value,
+                                                      energy_bin_edges.unit)
+        print("Found energy {0} in bin {1} with boundaries {2}.".format(energy,
+                                                                        energy_bin,
+                                                                        ss_energy_bin_edges))
+
+        # get data for the plot
+        ii = energy_bin
+        data = self.background[ii]
+        energy_bin_center = energy_bin_centers[ii]
+
+        # create plot
+        fig = plt.figure()
+        if ax is None:
+            ax = fig.add_subplot(111)
+
+        fig.set_size_inches(8., 8., forward=True)
+
+        image = ax.imshow(data.value,
+                          extent=extent.value,
+                          origin='lower', # do not invert image
+                          interpolation='nearest',
+                          norm=LogNorm(), # color log scale
+                          cmap='afmhot')
+
+        # set title and axis names
+        ax.set_title('Energy = [{0:.1f}, {1:.1f}) {2}'.format(energy_bin_edges[0].value,
+                                                              energy_bin_edges[1].value,
+                                                              energy_bin_edges.unit))
+        ax.set_xlabel('X / {}'.format(extent.unit))
+        ax.set_ylabel('Y / {}'.format(extent.unit))
+
+        # draw color bar
+        divider = make_axes_locatable(ax)
+        cax = divider.append_axes("right", size="5%", pad=0.05)
+        fig.colorbar(image, cax=cax, label='Bg rate / {}'.format(data.unit))
+
+        # save output
+        if save:
+            # save image in png file
+            filename = "cube_background_model_image"
+            filename += "{0}{1}.png".format(energy.value,
+                                            energy.unit)
+            print('Writing {}'.format(filename))
+            fig.savefig(filename)
+
+        return fig, ax, image
+
+    def plot_spectrum(self, det, ax=None, save=False):
+        """Plot spectra for the det bin containing the specified det (X, Y) pair.
+
+        Parameters
+        ----------
+        det : `~astropy.units.Quantity`
+            det (X,Y) pair of bin to plot the bg model
+        ax : `~matplotlib.axes.Axes`, optional
+            axes of the figure
+        save : `~bool`, optional
+            flag to turn on/off the saving of the figure in a png file
+
+        Returns
+        -------
+        fig : `~matplotlib.figure.Figure`
+            figure with image of bin of the bg model for the
+            selected det (X,Y) pair (if any), optional
+        ax : `~matplotlib.axes.Axes`
+            axes of the figure, optional
+        image : `~matplotlib.image.AxesImage`
```

Comment:
There's no need to return the `image` object.
It will be rare that a user needs it, but if they do, they can access it via the `ax` or `fig` object.

Actually, you can access `fig` from `ax` and `ax` from `fig`.
So in the interest of keeping the API simple, my suggestion would be to only take `ax` as input / output  ...

This is how I'd like all plotting functions / methods in Gammapy to work ... take an axes object as input / output.
Example: https://gammapy.readthedocs.org/en/latest/api/gammapy.time.plot_fermi_3fgl_light_curve.html